hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/StringUtils.java

```diff
@@ -494,9 +504,22 @@ public static Map<String, String> getTrimmedStringCollectionSplitByEquals(
     String[] trimmedList = getTrimmedStrings(str);
     Map<String, String> pairs = new HashMap<>();
     for (String s : trimmedList) {
+      if (s.length() == 0) {
+        continue;
+      }
       String[] splitByKeyVal = getTrimmedStringsSplitByEquals(s);
       if (splitByKeyVal.length == 2) {
-        pairs.put(splitByKeyVal[0], splitByKeyVal[1]);
+        boolean emptyKey = org.apache.commons.lang3.StringUtils.isEmpty(splitByKeyVal[0]);
+        boolean emptyVal = org.apache.commons.lang3.StringUtils.isEmpty(splitByKeyVal[1]);
+        if (!emptyKey && !emptyVal) {
+          pairs.put(splitByKeyVal[0], splitByKeyVal[1]);
+        } else {
+          throw new IllegalArgumentException(
```

Comment:
use Preconditions.checkArgument(!emptyKey && !emptyVal with its built in string formatting.